parquet/src/arrow/arrow_writer/levels.rs

```diff
@@ -1397,4 +1504,160 @@ mod tests {
 
         assert_eq!(&levels[1], &expected_level);
     }
+
+    #[test]
+    fn test_fixed_size_list() {
+        // [[1, 2], null, null, [7, 8], null]
+        let mut builder = FixedSizeListBuilder::new(Int32Builder::new(), 2);
+        builder.values().append_slice(&[1, 2]);
+        builder.append(true);
+        builder.values().append_slice(&[3, 4]);
+        builder.append(false);
+        builder.values().append_slice(&[5, 6]);
+        builder.append(false);
+        builder.values().append_slice(&[7, 8]);
+        builder.append(true);
+        builder.values().append_slice(&[9, 10]);
+        builder.append(false);
+        let a = builder.finish();
+
+        let item_field = Field::new("item", a.data_type().clone(), true);
+        let mut builder =
+            LevelInfoBuilder::try_new(&item_field, Default::default()).unwrap();
+        builder.write(&a, 1..4);
+        let levels = builder.finish();
+
+        assert_eq!(levels.len(), 1);
+
+        let list_level = levels.get(0).unwrap();
+
+        let expected_level = LevelInfo {
+            def_levels: Some(vec![0, 0, 3, 3]),
+            rep_levels: Some(vec![0, 0, 0, 1]),
```

Comment:
```suggestion
            rep_levels: Some(vec![0, 1, 0, 1]),
```
As written it is not correct

Reply:
I believe that the original is correct since the levels are generated by `builder.write(&a, 1..4)`.

So the levels should correspond to the 1st, 2nd, and 3rd rows:
- def `0`, rep `0` (row 1, null)
- def `0`, rep `0` (row 2, null)
- def `3`, rep `0` (row 3, child 0)
- def `3`, rep `1` (row 3, child 1)

Is that not right?